tests/AppBundle/Controller/AdherentControllerTest.php

```diff
@@ -207,6 +207,50 @@ public function testEditAdherentInterests()
         }
     }
 
+    public function testAdherentChangePassword()
+    {
+        $this->authenticateAsAdherent($this->client, 'carl999@example.fr', 'secret!12345');
+
+        $crawler = $this->client->request(Request::METHOD_GET, '/espace-adherent/mon-profil/changer-mot-de-passe');
+
+        $this->assertCount(1, $crawler->filter('input[name="adherent_change_password[old_password]"]'));
+        $this->assertCount(1, $crawler->filter('input[name="adherent_change_password[password][first]"]'));
+        $this->assertCount(1, $crawler->filter('input[name="adherent_change_password[password][second]"]'));
+
+        // Submit the profile form with invalid data
```

Comment:
The client uses a cookie to keep its connection state. So, even if your password is updated (in the database), you will be able to perform some new requests. Ok, if the password is updated, you will be unable to make the password change in the next step...

What do you think about splitting your test in parts:
- Submit empty form
- Submit valid new passwords but wrong old
- Submit valid old password but not same new ones
- Submit valid form

So, just add two steps in your case. This is a security feature, this is why I like to have wide tests.